compiler/src/dmd/statementsem.d

```diff
@@ -3313,6 +3306,12 @@ Statement statementSemanticVisit(Statement s, Scope* sc)
         tcs.tryBody = sc.tryBody;   // chain on the in-flight tryBody
         tcs._body = tcs._body.semanticScope(sc, null, null, tcs);
 
+        if (!global.params.useExceptions)
+        {
+            result = tcs._body;
+            return;
+        }
```

Comment:
(New) line 3392 would only be reached if some `catch` clause survived, so when moving the old check there, I think we'd be good, not having to change the semantics (i.e., NOT removing `catch (Throwable)` etc. anymore).